hatrac/model/storage/amazons3.py

```diff
@@ -106,7 +124,16 @@ def create_from_file(self, name, input, nbytes, content_type=None, content_md5=N
             headers['Content-Type'] = content_type
         if content_md5 is not None:
             headers['Content-MD5'] = content_md5
-        s3_key.set_contents_from_stream(input, headers=headers, replace=True, md5=md5, size=nbytes)
```

Comment:
Are you saying that set_contents_from_stream is missing in your boto version?  Or that it is buggy and uses chunked transfer encoding even if we specify a size keyword argument?  I think it is important for us to stream the hatrac request data straight through to S3 without buffering the whole entity.  The size should be known, so there should be no reason for chunked transfer encoding...